tests/BaseElementTest.php

```diff
@@ -401,4 +401,50 @@ public function testAbsoluteLink(string $class, string $element, ?string $link)
         $absoluteLink = $object->AbsoluteLink();
         $this->assertEquals($link, $absoluteLink);
     }
+
+    public function previewLinksProvider()
+    {
+        return [
+            // Element on Page
+            [
+                ElementContent::class,
+                'content1',
+                '/test-elemental/',
+            ],
+            // Element in DataObject WITHOUT PreviewLink or Link
+            [
+                TestElement::class,
+                'elementDataObject2',
+                null,
+            ],
+            // Element in DataObject WITH PreviewLink WITHOUT Link
+            [
+                TestElement::class,
+                'elementDataObject3',
+                'preview-link',
+            ],
+            // Element in DataObject WITH PreviewLink AND Link (different paths)
+            [
+                TestElement::class,
+                'elementDataObject4',
+                'base-link',
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider previewLinksProvider
+     */
+    public function testPreviewLink(string $class, string $elementIdentifier, ?string $link)
+    {
+        /** @var BaseElement $element */
+        $element = $this->objFromFixture($class, $elementIdentifier);
+
+        if ($link) {
+            $regex = '/^' . preg_quote($link . '?ElementalPreview=', '/') .'\d*#' . $element->getAnchor() . '$/';
+            $this->assertTrue(preg_match($regex, $element->PreviewLink()));
```

Comment:
Just small things preg_match(...) returns 1 (found), 0 (not found) or false (error).
assertTrue doesn't work with integers. 
Could you replace this with 
```php
            $result = preg_match($regex, $element->PreviewLink()) ? true : false;
            $this->assertTrue($result);
```
or cast to Boolean type explicitly 
```php $this->assertTrue((bool)preg_match($regex, $element->PreviewLink()));```

Reply:
Can do.

Reply:
Done. Good catch.